http-server/src/server.rs

```diff
@@ -754,7 +763,14 @@ async fn process_health_request(
 	middleware.on_response(request_start);
 
 	match data {
-		Some(resp) if success => Ok(response::ok_response(resp)),
+		Some(data) if success => {
+			let json: serde_json::Value =
```

Comment:
this is super annoying but no way around that with huge refactorings.

Reply:
I wonder whether something like this would be any better (likely faster I'd expect):

```rust
#[derive(Deserialize)]
struct RpcPayload<'a> {
    #[serde(borrow)]
    result: &'a serde_json::value::RawValue
}

let json: RpcPayload = serde_json::from_str(&data).expect("valid JSON-RPC response is valid JSON; qed");
Ok(response::ok_response(json.result.to_string())
```